File: desafio1/src/Eventos/dto/update-event.dto.ts
import { 
  IsString, 
  IsOptional, 
  IsDate, 
  IsNumber, 
  MinLength 
} from 'class-validator';
import { Type } from 'class-transformer';

export class UpdateEventDto {
  @IsOptional()
  @IsString()
  @MinLength(5)
  title?: string;

  @IsOptional()
  @IsString()
  @MinLength(10)
  description?: string;

  @IsOptional()
  @IsDate()
  @Type(() => Date)
  date?: Date;

  @IsOptional()
  @IsString()
  location?: string;

  @IsOptional()
  @IsNumber()
  lat?: number;

  @IsOptional()
  @IsNumber()
  lng?: number;

  @IsOptional()
  imageData?: Buffer;

  @IsOptional()
  @IsString()
  imageType?: string;

  @IsOptional()
  @IsString()
  imageName?: string;
}

File: flutter_desafio1/lib/features/events/models/update_event_dto.dart
class UpdateEventDto {
  final String? title;
  final String? description;
  final DateTime? date;
  final String? location;
  final double? lat;
  final double? lng;
 

  UpdateEventDto({
    this.title,
    this.description,
    this.date,
    this.location,
    this.lat,
    this.lng,
  });

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    
    // Solo incluir campos que no sean null
    if (title != null) data['title'] = title;
    if (description != null) data['description'] = description;
    if (date != null) data['date'] = date!.toIso8601String();
    if (location != null) data['location'] = location;
    if (lat != null) data['lat'] = lat;
    if (lng != null) data['lng'] = lng;
    
    return data;
  }
}